### PR TITLE
{data}[gompi/2022.10] netCDF-Fortran v4.6.0, netCDF v4.9.0, HDF5 v1.12.2, Szip v2.1.1

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
requires:
* #16825

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 7 out of 7 (4 easyconfigs in total)
bear-pg0211u15a.bear.cluster - Linux Ubuntu 20.04.2 LTS (Focal Fossa), x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.8.5
See https://gist.github.com/cce047459e4433f1f1cc1791cfe1102f for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 7 out of 7 (4 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/a9a985bf072e63c2ab85027b9189dcd6 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by branfosj** [comment]
We disabled one of the parts of `nc_test4_run_par_test` in #15959, but we are not even getting that far. This part of the tests runs really slowly for some - seeing if more cores on generoso helps.

**Comment by boegel** [comment]
@branfosj How do you think we should proceed here? Extend the timeout?

**Comment by branfosj** [comment]
> @branfosj How do you think we should proceed here? Extend the timeout?

We should try with a longer timeout - to see how long it takes for these tests to complete. I expect that we'll just keep increasing it for the tests in here - I supsect that the functionality these tests hit are a bad fit for certain types of filesystems.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3104.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/9a852e73965ea63f7b1099fff2ef353a for a full test report.

**Comment by boegel** [comment]
I have cancelled job `9852` that was running on `generoso` with relatex timeout (`ARGS='--timeout 100000'` in `pretestopts`), since it was already running for 12.5h.

Instead, I've added a new patch in d72129702b which disable one more test, because it's running excessively long on some types of filesystems.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3104.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/846087f611eb1ad12fbc4784aebf2512 for a full test report.

File: easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-gompi-2022.10.eb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name = 'netCDF'
+version = '4.9.0'
+
+homepage = 'https://www.unidata.ucar.edu/software/netcdf/'
+description = """NetCDF (network Common Data Form) is a set of software libraries
+ and machine-independent data formats that support the creation, access, and sharing of array-oriented
+ scientific data."""
+
+toolchain = {'name': 'gompi', 'version': '2022.10'}
```
Comment (boegel) [inline easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-gompi-2022.10.eb:9]: Note: version should be updated to `2022b` once #16961 is merged

**Comment by SebastianAchilles** [comment]
I am also seeing the timeout on my system. In my case I am running the test report in a container and the filesystem is mounted into the container, e.g. similar to:
```
docker run -it -v /data/easybuild/:/home/easybuild/.local/easybuild/ ghcr.io/easybuilders/rockylinux-9.0:latest
```
`/data` is a 4TB SATA HDD with an `ext4` filesystem. Using `--buildpath=/dev/shm/$USER/` doesn't make a difference.

I tried to nail down the tests that are talking so long. To get rid of the timeout I had to disable in addition to the already disabled tests the `@MPIEXEC@ -n 4 ./tst_parallel_zlib` and `@MPIEXEC@ -n 4 ./tst_parallel_compress` tests, e.g. I added the following to the patch `netCDF-4.9.0_skip-timeout-tests.patch`:

```
@@ -50,14 +50,14 @@
 # Only run these tests if HDF5 supports parallel filters (v1.10.2 and
 # later).
 if test "@HAS_PAR_FILTERS@" = "yes"; then
-    echo
-    echo "Parallel I/O test with zlib."
-    @MPIEXEC@ -n 4 ./tst_parallel_zlib
+#     echo
+#     echo "Parallel I/O test with zlib."
+#     @MPIEXEC@ -n 4 ./tst_parallel_zlib
 
     echo
     echo "Parallel I/O more tests with zlib and szip (if present in HDF5)."
     @MPIEXEC@ -n 1 ./tst_parallel_compress
-    @MPIEXEC@ -n 4 ./tst_parallel_compress
+#     @MPIEXEC@ -n 4 ./tst_parallel_compress
 fi
 
 echo
```

Skipping these tests also works on `jsc-zen2`, which is using an NFS filesystem.

**Comment by Micket** [comment]
I would have thought the `netCDF-4.9.0_skip-nasa-test.patch`  patch was enough. That seemed to be enough for netCDF-4.9.0-gompi-2022a.eb and netCDF-4.9.0-iimpi-2022a.eb

It did fix things for the person who originated this issue a while back https://github.com/easybuilders/easybuild-easyconfigs/issues/15959

If it's not enough, then we should apply this new patch to those easyconfigs as well.

**Comment by SebastianAchilles** [comment]
Yes, in the 2022a toolchain skipping only the `Parallel Performance Test for NASA` for `netCDF-4.9.0-gompi-2022a.eb` worked on `generoso`, `jsc-zen2`, and my container that I am using for test reports. For the version for 2022b more tests seem to take longer. The exact reason why the tests take longer is not clear for me.

**Comment by Micket** [comment]
So the difference stems from the compiler and MPI change, or the version bump of libxml2 and cURL.
I would suspect MPI if anything. 

Maybe it's not so clear cut that these failures should just be ignored?

**Comment by boegel** [comment]
closing in favor of @jfgrimm's #17107 (which just got merged)...

**Pull request closed without merging**